compiler-core/src/language_server/tests.rs

```diff
@@ -392,6 +393,34 @@ impl<'a> TestProject<'a> {
         }
     }
 
+    pub fn src_from_module_url(&self, url: &Url) -> Option<&str> {
+        let module_name: EcoString = url
+            .path_segments()?
+            .skip_while(|segment| *segment != "src")
+            .skip(1)
+            .join("/")
+            .trim_end_matches(".gleam")
+            .into();
+
+        if module_name == "app" {
+            return Some(self.src);
+        }
+
+        let find_module = |modules: &Vec<(&'a str, &'a str)>| {
+            modules
+                .iter()
+                .find(|(name, _)| *name == module_name)
+                .map(|(_, src)| *src)
+        };
+
+        find_module(&self.root_package_modules)
+            .or(find_module(&self.dependency_modules))
+            .or(find_module(&self.test_modules))
+            .or(find_module(&self.hex_modules))
+            .or(find_module(&self.dev_hex_modules))
+            .or(find_module(&self.indirect_hex_modules))
```

Comment:
This isn't lazy so it does way more work than it needs to due to not short circuiting.

What's the difference between all of these? Why isn't it always `root_package_modules` or `dependency_modules`?

Reply:
I'm not sure what the purpose of each one is in the `TestProject` but since one could jump to the definition of something defined in any of those modules I extended the search to each one of those.

I've made it lazy!

Reply:
Oh sorry I just realised this is in the test code, I'm a fool. Sorry about that